core/src/main/scala/org/apache/spark/network/netty/SparkTransportConf.scala

```diff
@@ -55,4 +69,27 @@ object SparkTransportConf {
       }
     })
   }
+
+  /**
+   * Separate threads configuration of driver and executor
+   * @param conf the [[SparkConf]]
+   * @param module the module name
+   * @param server if true, it's for the serverThreads. Otherwise, it's for the clientThreads.
+   * @return the configured number of threads, or -1 if not configured.
+   */
+  def getSpecificNumOfThreads(
+                               conf: SparkConf,
+                               module: String,
+                               server: Boolean): Int = {
```

Comment:
Using `boolean` to differentiate driver and executor seems not so straight forward, also if we want to extend this method to other scenarios not just "driver" and "executor", this method seems cannot satisfy, so can we make this method more clear by passing "role" through parameters?

Reply:
yes. Let me try.

Reply:
@jerryshao , before I committing the changes, I want to double-check with you about the "role" to make sure we are on the same page. 
There are other two common roles, master and worker, in addition to driver and executor. I can determine which role is by the "system name" parameter passed to RpcEnv.create which in turn invoke SparkTransportConf to reach out my modification area. We can, for example,

_val workerPattern = "(sparkWorker.*)".r
def determineRole(systemName: String): String = {
    systemName match {
      case SparkEnv.executorSystemName => "executor"
      case SparkEnv.driverSystemName => "driver"
      case "sparkMaster" => "master"
      case workerPattern(_) => "worker"
      case _ @ name => name
    }
  }_
to map system name to roles. 
Or we can set role specifically in spark configuration object when instantiate master, worker and executor, just like driver setting executor id to "driver". 
Then we can use "spark.$role.$module.io.*" to get threads configurations. 
Is it what you talked about the "role"?

Reply:
I don't think we have to be too intelligent here. What I mean is that instead of using "boolean" to differentiate driver and executor, maybe we could pass in specific "word" to make it more straight forward.

Reply:
@jerryshao , I see.  thanks.